crates/engine/util/src/reorg.rs

```diff
@@ -377,13 +378,13 @@ where
     // and 4788 contract call
     state.merge_transitions(BundleRetention::PlainState);
 
-    let outcome: ExecutionOutcome = ExecutionOutcome::new(
-        state.take_bundle(),
-        Receipts::from(vec![receipts]),
+    let outcome = ExecutionOutcome::new(
+        state.finalize(),
+        Receipts::<Receipt>::from(vec![receipts]),
```

Comment:
for some reason when I added the `finalize()`, the compiler wasn't able to infer the generic for `Receipts` anymore.

Reply:
Let's review this when we implement the new pattern for the `FinalizeExecution` trait. It may be that the change provides the compiler the additional context it needs.

Reply:
I updated the PR with the new pattern for `FinalizeExecution`, but the compiler still needs this hint on the `Receipt` type.